src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -297,27 +298,124 @@ private ImmutableArray<ISymbol> GetSymbolsForExpressionOrStatementContext()
                 ? _context.SemanticModel.LookupStaticMembers(_context.LeftToken.SpanStart)
                 : _context.SemanticModel.LookupSymbols(_context.LeftToken.SpanStart);
 
-            // filter our top level locals if we're inside a type declaration.
-            if (_context.ContainingTypeDeclaration != null)
-                symbols = symbols.WhereAsArray(s => s.ContainingSymbol.Name != WellKnownMemberNames.TopLevelStatementsEntryPointMethodName);
```

Comment:
we were realizing a filtered list over and over again.  Tweaked the filtering to be single pass.